src/libs/blueprint/conduit_blueprint_mesh_matset_xforms.cpp

```diff
@@ -129,43 +129,82 @@ namespace matset
 //-----------------------------------------------------------------------------
 void
-to_silo(const conduit::Node &n,
+to_silo(const conduit::Node &field,
+        const conduit::Node &matset,
         conduit::Node &dest,
         const float64 epsilon)
 {
     Node temp, data;
-    const DataType int_dtype = find_widest_dtype2(n, blueprint::mesh::default_int_dtypes);
-    const DataType float_dtype = find_widest_dtype2(n, blueprint::mesh::default_float_dtype);
+    const DataType int_dtype = find_widest_dtype2(matset,
+                                            blueprint::mesh::default_int_dtypes);
+    const DataType float_dtype = find_widest_dtype2(matset,
+                                            blueprint::mesh::default_float_dtype);
+    // if matset_values is not empty, we will
+    // apply the same xform to it as we do to the volume fractions.
+    const bool xform_matset_values = field.has_child("matset_values");
 
-    // Extract Material Set Metadata //
+    // NOTE: matset values are always treated as a float64.
+    // we could map to the widest int or float type in the future.
 
-    const bool mset_is_unibuffer = blueprint::mesh::matset::is_uni_buffer(n);
-    const bool mset_is_matdom = blueprint::mesh::matset::is_material_dominant(n);
+    // Extract Material Set Metadata //
+    const bool mset_is_unibuffer = blueprint::mesh::matset::is_uni_buffer(matset);
+    const bool mset_is_matdom = blueprint::mesh::matset::is_material_dominant(matset);
 
+    // setup the material map, which provides a map from material names
+    // to to material numbers
     Node matset_mat_map;
-    if(mset_is_unibuffer)
+    
+    // mset_is_unibuffer will always have the material_map, other cases
+    // it is optional. If not given, the map from material names to ids
+    // is implied by the order the materials are presented in the matset node
+    if(matset.has_child("material_map") )
     {
-        matset_mat_map.set_external(n["material_map"]);
+        // uni-buffer case provides the map we are looking for
+        matset_mat_map.set_external(matset["material_map"]);
     }
     else // if(!mset_is_unibuffer)
     {
-        std::vector<std::string> mat_vec = n["volume_fractions"].child_names();
-        std::sort(mat_vec.begin(), mat_vec.end());
-        for(int64 mat_index = 0; mat_index < (index_t)mat_vec.size(); mat_index++)
+        // material_map is implied, construct it here for use and output
+        NodeConstIterator vf_itr = matset["volume_fractions"].children();
+        index_t mat_index = 0;
+        while(vf_itr.has_next())
         {
-            const std::string &mat_name = mat_vec[mat_index];
+            vf_itr.next();
+            std::string curr_mat_name = vf_itr.name();
+
             temp.set_external(DataType::int64(1), &mat_index);
-            temp.to_data_type(int_dtype.id(), matset_mat_map[mat_name]);
+            temp.to_data_type(int_dtype.id(), matset_mat_map[curr_mat_name]);
```

Comment:
I think this look could also be simplified with `vf_itr.index()`.

Reply:
FYI: Good suggestion, I made this change.